frontend/src/components/WildcardSelect.vue

```diff
@@ -0,0 +1,215 @@
+You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+-->
+
+<template>
+  <div class="d-flex flex-row">
+    <v-select
+      class="selectClass"
+      color="cyan darken-2"
+      item-color="cyan darken-2"
+      :label="wildcardSelectLabel"
+      :items="wildcardSelectItemObjects"
+      return-object
+      v-model="wildcardSelectedValue"
+      :error-messages="getErrorMessages('wildcardSelectedValue')"
+      @input="onInput"
+      @blur="$v.wildcardSelectedValue.$touch()"
+      :hint="wildcardSelectHint"
+      persistent-hint
+    >
+      <template v-slot:item="{ item }">
+        <v-list-item-content>
+          <v-list-item-title>
+            <span v-if="item.startsWithWildcard">&lt;prefix&gt;</span>
+            <span>{{item.text}}</span>
+            <span v-if="item.endsWithWildcard">&lt;suffix&gt;</span>
+          </v-list-item-title>
+        </v-list-item-content>
+      </template>
+    </v-select>
+    <v-text-field
+      v-if="wildcardSelectedValue.isWildcard"
+      :class="textFieldClass"
+      color="cyan darken-2"
+      :label="wildcardTextFieldLabel"
+      v-model="wildcardVariablePart"
+      :error-messages="getErrorMessages('wildcardVariablePart')"
+      @input="onInput"
+      @blur="$v.wildcardVariablePart.$touch()"
+      ></v-text-field>
+  </div>
+</template>
+
+<script>
+import startsWith from 'lodash/startsWith'
+import endsWith from 'lodash/endsWith'
+import map from 'lodash/map'
+import forEach from 'lodash/forEach'
+import trim from 'lodash/trim'
+import { getValidationErrors } from '@/utils'
+import { required, requiredIf } from 'vuelidate/lib/validators'
+
+const validations = {
+  wildcardVariablePart: {
+    required: requiredIf(function () {
+      return this.wildcardSelectedValue.isWildcard
+    })
+  },
+  wildcardSelectedValue: {
+    required
+  }
+}
+
+export default {
+  name: 'wildcard-select',
+  props: {
+    wildcardSelectItems: {
+      type: Array
+    },
+    wildcardSelectLabel: {
+      type: String
+    },
+    value: {
+      type: String
+    }
+  },
+  data () {
+    return {
+      wildcardVariablePart: undefined,
+      wildcardSelectedValue: {},
+      valid: undefined
+    }
+  },
+  computed: {
+    validationErrors () {
+      return {
+        wildcardVariablePart: {
+          required: `${this.wildcardTextFieldLabel} is required`
+        },
+        wildcardSelectedValue: {
+          required: `${this.wildcardSelectLabel} is required`
+        }
+      }
+    },
+    textFieldClass () {
+      return this.wildcardSelectedValue.startsWithWildcard ? 'textFieldStartClass' : 'textFieldEndClass'
```

Comment:
just to make sure: a value like `*foo*` is not possible, right (starts and ends with wildcard)?

Reply:
I don't think so, at least not supported by the dashboard. @timuthy is this supported by gardener?

Reply:
No, also not supported by Gardener.